tests/tcmpkeys.c

```diff
@@ -44,6 +44,14 @@ static EVP_PKEY *load_key(const char *uri)
         exit(EXIT_FAILURE);
     }
 
+    if (strncmp(uri, "pkcs11:", 7)) {
+        /* public file */
+        if (OSSL_STORE_expect(store, OSSL_STORE_INFO_PUBKEY) != 1) {
+            fprintf(stderr, "Failed to expect Public Key File\n");
+            exit(EXIT_FAILURE);
+        }
+    }
```

Comment:
I do not think this is correct. The uris in the test below are `EDPRIURI` or `EDPUBURI`, so we should be interested in getting the private key when requested (which is broken in openssl 3.1). I think this is just a workaround for the OpenSSL issue openssl/openssl#22246 (and openssl/openssl#21219) as in most of the useful cases, we are interested in the private key to be able to use it for something sensible.

This significantly limits the testing. I do not mind adding another option in the tcmpkeys to check we get public key, but we should be able to get a private key too.

Reply:
I pushed a commit that should address these concerns.